Enhancements (Docker, PyLint, PyCodeStyle, Travis-CI)

farhan-arshad-dev opened the pull request:
### Project code and feature Enhancements
- Integration of docker, travis-ci, and PyLint, PyCodeStyle.
- Resolve PyLint, PyCodeStyle reported issues.
- Added Unit tests for the project feature. 
- Using TDD approach to update the admin page for user profile model. 
- Configure Postgresql.

Commit:
Setup docker and docker compose

Commit:
Added flake8 and Travis-CI configrations.

Commit:
Add basic unit test

Commit:
Update README.md for the python package

Commit:
Resolve flake8 lint issues

Commit:
Write test using TDD

- First write test to subtract two number.
- After that create the method to subtract the two numbers

Commit:
Added unit test for custom user model

Commit:
Updated admin to support custom user model.

Commit:
Added postgresql configurations

- docker-compose configs.
- Dockerfile configs.
- Update project settings to connect with the postgresql service.

Commit:
Update path for basic unit tests

Commit:
Add postgres database unit test

- Added wait_for_db management command

Commit:
Resolve docker-compose up command issue

- Fix ERROR: for app  Cannot start service app: OCI runtime create failed: container_linux.go:345: starting container process caused "exec: \"#\": executable file not found in $PATH": unknown
ERROR: Encountered errors while bringing up the project.
- Update README

Commit:
Update Django version.

waheedahmed commented on docker-compose.yml:
```diff
@@ -0,0 +1,47 @@
+      - db
+
+  # service for database called 'db'  
+  db: # name of the service for the database
```
Move `db` at the top then you don't need `wait_for_db` command I believe.

waheedahmed commented on .travis.yml:
```diff
@@ -0,0 +1,15 @@
+
+script:
+  # run the docker container and then execute the test and linting tool
+  - docker-compose run app sh -c "python recipe_management_project/manage.py test && flake8"
```
Always add a new line at end of file, add here and in other files too where missing.

Commit:
Added new line at the end of files

Commit:
Remove wait_for_db command dependency

- Move db service above from the app service.
- Remove wait_for_db python command.

waheedahmed commented on src/recipe_management_project/api/management/commands/wait_for_db.py:
```diff
@@ -0,0 +1,34 @@
+# 'time' use for the application sleep for some sec
```
Remove this command and related tests if not needed.

waheedahmed commented on src/recipe_management_project/api/views.py:
```diff
@@ -103,15 +103,22 @@ def perform_create(self, serializer):
-            raise NotAcceptable("You already following {} user".format(serializer.validated_data['following_to']))
-        
+        # check user is trying to follow already following user
+        is_already_following = following_models.FollowingModel.objects.\
```
nit: 
```
is_already_following = following_models.FollowingModel.objects.filter(
    following_to=following_to.id,
    following_by=self.request.user.id
)
```

waheedahmed commented on src/recipe_management_project/api/views.py:
```diff
@@ -103,15 +103,22 @@ def perform_create(self, serializer):
+
     def get_queryset(self):
-        return following_models.FollowingModel.objects.filter(following_by=self.request.user.id)
+        return following_models.FollowingModel.objects.\
```
Same here.

waheedahmed reviewed (changes requested):
Generally looks good, some small nits. Integerate Pylint and PEP8 and run them in travis build.

Commit:
Suggested changes

farhan-arshad-dev commented:
> Generally looks good, some small nits. Integerate Pylint and PEP8 and run them in travis build.

@waheedahmed after Integration of `Pylint`, are we need `flake8`?

waheedahmed commented:
@farhan-arshad-dev No, but do integerate PEP8 and also check `flake8` is correctly running during your build and reporting the voilations correctly if present.

waheedahmed commented:
I don't see any log present in your build saying anything related to if flake8 command ran successfully or not.

farhan-arshad-dev commented:
> @farhan-arshad-dev No, but do integerate PEP8 and also check `flake8` is correctly running during your build and reporting the voilations correctly if present.

flake8 is working fine, its show report if there is any error in the code.

farhan-arshad-dev commented:
> > Generally looks good, some small nits. Integerate Pylint and PEP8 and run them in travis build.
> 
> @waheedahmed after Integration of `Pylint`, are we need `flake8`?

Are we need to integrate the `Pylint` and `PEP8` (renamed as `pycodestyle`) in docker container?

Commit:
Integration of Pylint & PyCodeStyle (PEP8)

- Add pylint django dependency
- Remove flake8
- Added Configs pylintrc for pylint and setup.cfg for pycodestyle
- Update gitignore
- Fix pylint and pycodestyle issue

Commit:
Resolve Pylint issues

- import-error
- too-many-ancestors
- no-self-use
- abstract-method
- Added .editorconfig for vscode
